baseapp/internal/protocompat/protocompat.go

```diff
@@ -235,6 +233,20 @@ func ResponseFullNameFromMethodDesc(sd *grpc.ServiceDesc, method grpc.MethodDesc
 
 // since proto.Merge breaks due to the custom cosmos sdk any, we are forced to do this ugly setPointer hack.
 // ref: https://github.com/cosmos/cosmos-sdk/issues/22779
-func setPointer(dst, src any) {
-	reflect.ValueOf(dst).Elem().Set(reflect.ValueOf(src).Elem())
+func setPointer(dst, src any) error {
```

Comment:
It might be worthwhile to test this function to ensure no regressions.  Is there a function where it is already indirectly exercised?

Reply:
yes, the failing tests :D